# Generated by Django 4.1.2 on 2023-11-01 19:05

from django.conf import settings
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
        ('deuldou', '0013_remove_liste_invitation_created_by_and_more'),
    ]

    operations = [
        migrations.CreateModel(
            name='ListeContacts',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('nom', models.CharField(max_length=100)),
                ('contacts', models.ManyToManyField(to='deuldou.contact')),
                ('user', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='get_liste_contacts', to=settings.AUTH_USER_MODEL)),
            ],
            options={
                'ordering': ['nom'],
            },
        ),
        migrations.DeleteModel(
            name='Liste_contacts',
        ),
        migrations.AddConstraint(
            model_name='listecontacts',
            constraint=models.UniqueConstraint(fields=('user', 'nom'), name='liste unique', violation_error_message='Cette liste existe déjà'),
        ),
    ]
